src/reducer.js

```diff
@@ -52,16 +52,43 @@ function recursiveClean (value, model) {
   let isValueArray = Array.isArray(value)
   let output = isValueArray ? [] : {}
   let iteratorFn = isValueArray ? _.forEach : _.forIn
+
   iteratorFn(value, (subValue, key) => {
-    const notEmpty = !_.isEmpty(subValue) || subValue instanceof File
-    if (Array.isArray(subValue) && notEmpty) {
-      output[key] = recursiveClean(subValue, _.get(model, 'items'))
-    } else if (!(subValue instanceof File) && _.isObject(subValue) && notEmpty) {
-      output[key] = recursiveClean(subValue, _.get(model, 'properties.' + key))
-    } else if (notEmpty || _.isNumber(subValue) || typeof subValue === 'boolean' || subValue instanceof Boolean) {
-      output[key] = subValue
+    const notEmpty = !_.isEmpty(subValue) ||
+      _.isNumber(subValue) ||
+      typeof subValue === 'boolean' ||
+      subValue instanceof Boolean ||
+      subValue instanceof File
+
+    let subModel
+
+    // arrays have their model props nested further than objects
+    if (Array.isArray(subValue)) {
```

Comment:
Thanks for trying to clean this up but I'm not sure if this reads better IMO. It is slightly more complex if you count the number of branches.

Reply:
I did it so I could get a handle on the subModel for the autoclean property - otherwise it would only work if the prop was set on a non-empty parent.

Reply:
Ah